javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -506,6 +487,50 @@ && shouldGeneratePagingMethods()) {
             .collect(Collectors.toList());
     }
 
+    private enum ParameterTransformationType {
+        FLATTEN,
+        GROUPING
+    }
+
+    private void processParameterTransformations(
+            ParameterTransformationType type,
+            List<MethodTransformationDetail> methodTransformationDetails,
+            Set<Parameter> originalParameters,
+            Parameter parameter, ClientMethodParameter clientMethodParameter,
+            boolean isProtocolMethod) {
```

Comment:
Changes to this class is mostly refactor.

The transformation moved here, so that it can be called on same parameter to either GROUPING or FLATTEN.

The case is that the body properties get flattened to parameters, and then these parameters get grouped to an option bag.